slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/dao/SignedAttestationsDao.java

```diff
@@ -12,22 +12,44 @@
  */
 package tech.pegasys.web3signer.slashingprotection.dao;
 
+import java.util.List;
 import java.util.Optional;
 import java.util.stream.Stream;
 
+import org.apache.tuweni.bytes.Bytes;
 import org.apache.tuweni.units.bigints.UInt64;
 import org.jdbi.v3.core.Handle;
 
 public class SignedAttestationsDao {
 
-  public Optional<SignedAttestation> findExistingAttestation(
-      final Handle handle, final int validatorId, final UInt64 targetEpoch) {
+  public List<SignedAttestation> findAttestationsForEpochWithDifferentSigningRoot(
+      final Handle handle,
+      final int validatorId,
+      final UInt64 targetEpoch,
+      final Bytes signingRoot) {
+    return handle
+        .createQuery(
+            "SELECT validator_id, source_epoch, target_epoch, signing_root "
+                + "FROM signed_attestations WHERE (validator_id = ? AND target_epoch = ?) AND (signing_root <> ? OR signing_root IS NULL)")
```

Comment:
you may also make the signing root condition simpler with `...AND signing_root NOT IN (?)`. Hopefully these should pick both null and different signing roots.